cmd/segment-reaper/detect.go

```diff
@@ -44,5 +83,144 @@ func cmdDetect(cmd *cobra.Command, args []string) (err error) {
 	defer func() {
 		err = errs.Combine(err, db.Close())
 	}()
+
+	file, err := os.Create(detectCfg.File)
+	if err != nil {
+		return errs.New("error creating result file: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, file.Close())
+	}()
+
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+	headers := []string{
+		"ProjectID",
+		"SegmentIndex",
+		"Bucket",
+		"EncodedEncryptedPath",
+		"CreationDate",
+	}
+	err = writer.Write(headers)
+	if err != nil {
+		return err
+	}
+
+	objects := make(ObjectsMap)
+	inlineSegments := 0
+	lastInlineSegments := 0
+	remoteSegments := 0
+	lastProjectID := ""
+
+	// iterate method is serving entries in sorted order
+	err = db.Iterate(ctx, storage.IterateOptions{Recurse: true},
```

Comment:
Can we use metainfo loop for this, alternatively refactore metainfo.Loop such that we can use it without running it as a chore?

Reply:
I assumed that we don't want to invest in this as it will be just a temporary solution but I can check if it will be easy to reuse metainfo.Loop code here.

Reply:
I think you create a helper func in `metainfo` of something `IterateDatabase(ctx context.Context, db PointerDB, observers ...Observer)`